src/Std/Init/Option.lean

```diff
@@ -0,0 +1,79 @@
+/-
+Copyright (c) 2024 Lean FRO, LLC. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Markus Himmel
+-/
+prelude
+import Init.Data.Option.Lemmas
+
+namespace Option
+
+@[simp] theorem isSome_map (α : Type u) (β : Type v) (f : α → β) (o : Option α) :
+    (o.map f).isSome = o.isSome := by
+  cases o <;> simp
+
+@[simp] theorem get!_none [Inhabited α] : (none : Option α).get! = default := rfl
+@[simp] theorem get!_some [Inhabited α] {a : α} : (some a).get! = a := rfl
+
+theorem get_eq_get! [Inhabited α] : (o : Option α) → {h : o.isSome} → o.get h = o.get!
+  | some _, _ => rfl
+
+theorem get_eq_getD {fallback : α} : (o : Option α) → {h : o.isSome} → o.get h = o.getD fallback
+  | some _, _ => rfl
+
+theorem some_get! [Inhabited α] : (o : Option α) → o.isSome → some (o.get!) = o
+  | some _, _ => rfl
+
+theorem get!_eq_getD_default [Inhabited α] (o : Option α) : o.get! = o.getD default := rfl
+
+def or : Option α → Option α → Option α
+  | o@(some _), _ => o
+  | none, o => o
```

Comment:
We've got by with `orElse` so far. Would this suffice for your purposes?

If not, please add a doc-string.

Reply:
I think we want this in the interest of provding a very deep API for all of the basic types. Of course you can get by with `orElse`, but having `or` is just a bit more convenient sometimes. Rust has both (Java only has `orElse`).  It's very canonical, it makes `Option α` into a monoid for all `α`.

Docstrings will follow as promised :)

Reply:
See #4600.